source/slang/slang-ir-link.cpp

```diff
@@ -1519,6 +1520,146 @@ static void diagnoseUnresolvedSymbols(TargetRequest* req, DiagnosticSink* sink,
+                    // ignore the def
+                    break;
+                }
+                case kIROp_Call:
+                { 
+                    builder.setInsertBefore(user);
+                    auto fieldAddress = builder.emitFieldAddress(
+                        builder.getPtrType(_dataField->getFieldType()),
+                        instToSwitch,
+                        _dataField->getKey()
+                        );
+                    auto elementAddr = builder.emitElementAddress(
+                        builder.getPtrType(builder.getUIntType()),
+                        fieldAddress,
+                        builder.getIntValue(builder.getIntType(), currOffset/4));
+
+                    user->setOperand(1, elementAddr);
+                    auto funcTypeInst = (user->getOperand(0));
+                    auto funcType = funcTypeInst->getFullType();
+
+                    auto paramReplacment = builder.getInOutType(builder.getUIntType());
+                    funcType->getOperand(1)->replaceUsesWith(paramReplacment);
+                    builder.addForceInlineDecoration(funcTypeInst);
+
+                    break;
+                }
+                }
+            }
+            if (i->typeUse.usedValue->getOp() == kIROp_GLSLAtomicUintType)
+            {
+                i->removeAndDeallocate();
+            }
+        }
+    }
+}
+
+void tryGLSLTypeReplacmentProcessing(IRSpecContext* context, TargetProgram* targetProgram, IRModule* irModule)
+{
+    // this is an issolated boolean operation since this operation can be moved to different
+    // parts of Slang and may likley need to be moved at some point
+    if (targetProgram->getOptionSet().getBoolOption(CompilerOptionName::AllowGLSL))
+    {
+        /// do 1 pass and get all data needed to process GLSL inst's that need attention
+        Dictionary<int, List<IRInst*>> bindingToInstMapUnsorted;
+        for (auto inst : irModule->getGlobalInsts())
+        {
+            if (inst->typeUse.usedValue)
+            {
+                switch (inst->typeUse.usedValue->getOp())
+                {
+                case kIROp_GLSLAtomicUintType:
+                {
+                    // Atomic_uint are supported by GLSL through converting to a different type (compiler ext)
+                    // not supported by SPIR-V->VK, means we must convert the type ourselves, else SPIR-V + 
+                    // HLSL is going to be impossible to emit as a target
+
+                    // we fetch all global instances to pre-process
+                    auto layout = inst->findDecoration<IRLayoutDecoration>()->getLayout();
+                    auto layoutVal = as<IRVarOffsetAttr>(layout->getOperand(1));
+                    assert(layoutVal != nullptr);
+                    bindingToInstMapUnsorted.getOrAddValue(uint32_t(layoutVal->getOffset()), List<IRInst*>()).add(inst);
+                    break;
+                }
+                };
+            }
+        }
+        ///
```

Comment:
Please remove this comment.

Reply:
I suppose I will explain this topic just for future reference; GLSL has lots of hacks and random things which are removed if GLSL->VK is the target. GLSL also has lots of odd styling which makes little sense when translating GLSL->[SPIRV|CUDA|HLSL], as a result there will need to be some styling at some point to make this section less cluttered (since it likely will be used more)